osu.Game/Overlays/SocialOverlay.cs

```diff
@@ -62,65 +51,174 @@ public SocialOverlay()
+                if (Header.Tabs.Current.Value != SocialTab.AllPlayers && sortCriteria != (SocialSortCriteria)Header.Tabs.Current.Value)
+                    Header.Tabs.Current.Value = SocialTab.AllPlayers;
+
+                Scheduler.AddOnce(updateSearch);
+            };
+
+            Filter.DisplayStyleControl.DisplayStyle.ValueChanged += recreatePanels;
+            Filter.DisplayStyleControl.Dropdown.Current.ValueChanged += sortOrder => Scheduler.AddOnce(updateSearch);
+
+            //currentQuery.ValueChanged += v =>
+            //{
+            //    queryChangedDebounce?.Cancel();
+
+            //    if (string.IsNullOrEmpty(v))
+            //        Scheduler.AddOnce(updateSearch);
+            //    else
+            //        queryChangedDebounce = Scheduler.AddDelayed(updateSearch, 500);
+            //};
+
+            //currentQuery.BindTo(Filter.Search.Current);
         }
 
         [BackgroundDependencyLoader]
         private void load(APIAccess api)
         {
-            if (Users == null)
-                reloadUsers(api);
+            this.api = api;
+            api.Register(this);
         }
 
-        private void reloadUsers(APIAccess api)
+        private void recreatePanels(PanelDisplayStyle displayStyle)
         {
-            Users = null;
+            clearPanels();
 
-            // no this is not the correct data source, but it's something.
-            var request = new GetUsersRequest();
-            request.Success += res =>
+            if (Users == null)
+                return;
+
+            var newPanels = new FillFlowContainer<SocialPanel>
             {
-                Users = res.Select(e => e.User);
-                loading.Hide();
+                RelativeSizeAxes = Axes.X,
+                AutoSizeAxes = Axes.Y,
+                Spacing = new Vector2(10f),
+                Margin = new MarginPadding { Top = 10 },
+                ChildrenEnumerable = Users.Select(u =>
+                {
+                    SocialPanel panel;
+                    switch (displayStyle)
+                    {
+                        case PanelDisplayStyle.Grid:
+                            panel = new SocialGridPanel(u)
+                            {
+                                Anchor = Anchor.TopCentre,
+                                Origin = Anchor.TopCentre
+                            };
+                            break;
+                        default:
+                            panel = new SocialListPanel(u);
+                            break;
+                    }
+                    panel.Status.BindTo(u.Status);
+                    return panel;
+                })
             };
 
-            api.Queue(request);
+            LoadComponentAsync(newPanels, f =>
+            {
+                if(panels != null)
+                    ScrollFlow.Remove(panels);
+
+                ScrollFlow.Add(panels = newPanels);
+            });
+        }
+
+        private void clearPanels()

```

Comment:
To fix the animation issue you were having, you probably want to return `FadeOut`'s `TransformSequence` and attach the `FadeIn` of the next panels inside a `.Then.Schedule((() => {  })` or similar.

Reply:
Nope. I can get a sequence but the content of `Schedule` after a `Then` is never called.
I thought it's because the old panels are Expiring and even set to null but that doesn't seem to affect anything.
Result is that I still can't get it to work.